gtsam/geometry/Line3.h

```diff
@@ -0,0 +1,178 @@
+     * The tangent space for the rotation of a line is only two dimensional -
+     * rotation about x and y
+     * @param v: increment in tangent space
+     * @param H: Jacobian of retraction with respect to the increment
+     * @return: resulting line after adding the increment and mapping to the manifold
+     */
+    Line3 retract(const Vector4 &v, OptionalJacobian<4,4> H = boost::none) const;
+
+    /**
+     * The localCoordinates method is the inverse of retract and finds the difference
+     * between two lines in the tangent space.
+     * @param q Line3 on manifold
+     * @param H OptionalJacobian of localCoordinates with respect to line
+     * @return Vector4 difference in the tangent space
+     */
+    Vector4 localCoordinates(const Line3 &q, OptionalJacobian<4,4> H = boost::none) const;
+
+    /**
+     * Rotation of line accessor
+     * @return Rot3
+     */
+    Rot3 R() const
+    {
+        return R_;
+    }
+
+    /**
+     * Accessor for a, b
+     * @return Vector2(a, b)
+     */
+    Vector2 V() const
+    {
+        return Vector2(a_, b_);
+    }
+
+    /**
+     * Print R, a, b
+     * @param s: optional starting string
+     */
+    void print(const std::string &s = "") const;
+
+    /**
+     * Check if two lines are equal
+     * @param l2 - line to be compared
+     * @param tol : optional tolerance
+     * @return boolean - true if lines are equal
+     */
+    bool equals(const Line3 &l2, double tol = 10e-9) const;
+
+    /**
+     * Projecting a line to an image plane
+     * @param Dline: OptionalJacobian of projected line with respect to this line
+     * @return Point3 - projected line in image plane
+     */
+    Point3 project(OptionalJacobian<3, 4> Dline = boost::none) const;
+};
+
+template <>
+struct traits<Line3> : public internal::Manifold<Line3> {};
+
+template <>
+struct traits<const Line3> : public internal::Manifold<Line3> {};
+
+// For expression factor graphs
+typedef Expression<Line3> Line3_;
+
+/**
+ * Function projects input line to image plane
+ * @param L input line
+ * @param Dline OptionalJacbian of projected line with respect to input line
+ * @return Point3 - projected line
+ */
+Point3 projectLine(const Line3 &L,
+        OptionalJacobian<3, 4> Dline = boost::none);
+
+/**
+ * Transform a line from world to camera frame
+ * @param p - Pose3 of camera in world frame
+ * @param l - Line3 in world frame
+ * @param Dpose - OptionalJacobian of transformed line with respect to p
+ * @param Dline -  OptionalJacobian of transformed line with respect to l
+ * @return Transformed line in camera frame
+ */
+Line3 transformTo(const Pose3 &p, const Line3 &l,
+                    OptionalJacobian<4, 6> Dpose = boost::none,
+                    OptionalJacobian<4, 4> Dline = boost::none);
+
+/**
+ * Expression method for projection
+ * @param line Line3_ expression
+ * @return Point3_ projected line
+ */
+inline Point3_ projectLineExpression(const Line3_& line)
```

Comment:
Not here. It's easy to make an expression straight from the method declaration. I think something akin to

```
Point3_ project_(line_, &Line3::project);
```

should work, hence this stuff here is redundant. If it goes anywhere, probably in geometry/expressions.h